packages/react/src/components/Authenticator/shared/__tests__/FormFields.test.tsx

```diff
@@ -0,0 +1,76 @@
+import { render, screen } from '@testing-library/react';
+import * as React from 'react';
+
+import { defaultFormFieldOptions } from '@aws-amplify/ui';
+import { FormField, FormFieldProps } from '../FormField';
+import * as UseAuthenticatorModule from '../../hooks/useAuthenticator';
+import * as UseStableIdModule from '../../../../primitives/utils/useStableId';
+
+const usernameFormFieldProps: FormFieldProps = {
+  ...defaultFormFieldOptions.username,
+  name: 'username',
+  label: 'Username',
+};
+
+const useAuthenticatorSpy = jest.spyOn(
+  UseAuthenticatorModule,
+  'useAuthenticator'
+);
```

Comment:
I found `jest.spyOn` easier to work with than `jest.mock` when we want its return value to change between tests.